lisa/testsuite.py

```diff
@@ -752,6 +752,16 @@ def start(
                     log=case_log,
                 )
 
+            nodes = environment.nodes
```

Comment:
Move into `__run_case` after test passed, and let the `handle_exception` proceed the raised exception.